spec.html

```diff
@@ -35418,16 +35853,37 @@ <h1>
         </emu-alg>
         <emu-grammar>Atom :: `.`</emu-grammar>
         <emu-alg>
-          1. Let _A_ be the CharSet of all characters.
+          1. Let _A_ be _AllCharacters_.
           1. If _DotAll_ is not *true*, then
             1. Remove from _A_ all characters corresponding to a code point on the right-hand side of the |LineTerminator| production.
           1. Return ! CharacterSetMatcher(_A_, *false*, _direction_).
         </emu-alg>
         <emu-grammar>Atom :: CharacterClass</emu-grammar>
         <emu-alg>
           1. Let _cc_ be CompileCharacterClass of |CharacterClass|.
-          1. Return ! CharacterSetMatcher(_cc_.[[CharSet]], _cc_.[[Invert]], _direction_).
-        </emu-alg>
+          1. Let _cs_ be _cc_.[[CharSet]].
+          1. If _UnicodeSets_ is *false* or if every element of _cs_ consists of a single character, return ! CharacterSetMatcher(_cs_, _cc_.[[Invert]], _direction_).
```

Comment:
These will never get to the runtime semantics because the static semantics have this:

CharacterClass :: [ ^ ClassContents ]
- It is a Syntax Error if MayContainStrings of the ClassContents is true.

Reply:
Perfect, thanks!